File: solutions/patterns/star-patterns/11-hollow-full-pyramid/solution.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Enter the N value: ";
    cin >> n;

    cout << "\nPrinting the pattern...\n\n";

    for (int row = 1; row <= n; row++) {
        // Print leading spaces
        for (int col = 1; col <= n - row; col++) {
            cout << " ";
        }

        if (row == 1) {
            // First row, single star
            cout << "*";
        } else if (row == n) {
            // Last row, full stars with space
            for (int star = 1; star <= n; star++) {
                cout << "* ";
            }
        } else {
            // Middle rows
            cout << "*";  // Left star

            // Inner spaces
            for (int space = 0; space < (2 * row - 3); space++) {
                cout << " ";
            }

            cout << "*";  // Right star
        }

        cout << endl;
    }

    return 0;
}

File: solutions/patterns/star-patterns/11-hollow-full-pyramid/solution.js
let n = 5

for (let i = 0; i < n; i++) {
    let row = ""
    for (let j = 0; j < n - i - 1; j++) {
        row += " "
    }
    for (let k = 0; k < 2 * i + 1; k++) {
        if (k == 0 || k == 2 * i || (i == n - 1 && k % 2 == 0)) {
            row += "*";
        } else {
            row += " ";
        }
    }
    console.log(row)
}